src/clj/fluree/db/reasoner.cljc

```diff
@@ -218,24 +235,42 @@
             (recur r db**))
           db*)))))
 
+(defn deduplicate-raw-rules
+  [raw-rules]
+  (let [rule-ids (map first raw-rules)
+        duplicate-ids (filter #(< 1 (last %)) (frequencies rule-ids))]
+    (reduce (fn [rules [duplicate-id occurances]]
+              (let [grouped-rules (group-by #(= duplicate-id (first %)) rules)]
+                (loop [suffix occurances
+                       rules-to-update (get grouped-rules true)
+                       updated-rules-list (get grouped-rules false)]
+                  (if (empty? rules-to-update)
+                    updated-rules-list
+                    (let [updated-rule [(str duplicate-id suffix) (last (first rules-to-update))]]
+                      (recur (dec suffix) (rest rules-to-update) (conj updated-rules-list updated-rule)))))))
+            raw-rules duplicate-ids)))
+
 (defn reason
-  [db methods graph-or-db {:keys [max-fuel reasoner-max]
-                           :or   {reasoner-max 10} :as _opts}]
+  [db methods
+   {:keys [rule-graphs rule-dbs]}
```

Comment:
If we do support a list of sources here, I don't think we should require special key syntax to distinguish them. By requiring this special map syntax, you are going to require our users to have to look up this special syntax likely every time they use the API - even if they are only passing in a single rule set (which is the most common case).

 Instead, you could easily call `db?` on each item and do the right thing without needing any special syntax at all.